weatherreport.cpp

```diff
@@ -36,6 +37,25 @@ namespace WeatherSpace
             return 52;
         }
     };
+
+    class HighPrecipitationStub : public IWeatherSensor {
```

Comment:
The name `HighPrecipitationStub` clearly conveys what kind of weather it is stubbing. Change the older `SensorStub` also to such a specific name.
Also, you have chosen to represent different stubs as different classes. Any reason you did not go for different instances of one class?